src/NumFieldOrd/NfOrd/Clgp/Saturate.jl

```diff
@@ -28,12 +29,21 @@ function mod_p(R::Vector{FacElem{AbsSimpleNumFieldElem, AbsSimpleNumField}}, Q::
     end
     x = rand(F)
   end
-  y = x
-  for i = 1:pp-1
-    dl[y] = i
-    y *= x
+  if nbits(pp) < 19 # seems to be a reasonable cutoff
+    y = one(F)
+    for i = 0:pp-1
+      dl[y] = i
+      y *= x
+    end
+  end
+  ma = Vector{Int}(undef, length(R))
+  for i in 1:length(R)
+    imgd = image(mF1, R[i], D[i], cached, pp)^e
+    ma[i] = get!(dl, imgd) do
+      ma[i] =  Hecke.baby_step_giant_step(x, pp, imgd, dl) % p
```

Comment:
it feels like one of the `ma[i] = ` should be superflous. dies the `get` also update the `dl` cache?

Reply:
oops, you are right